### PR TITLE
added lin_alg support for cosine similarity

### PR DESCRIPTION
**Pull request by DhavalTaunk08**

### CODE REVIEW

File: lib/src/vector_db.dart [hunk 1]
```diff
@@ -1,14 +1,20 @@
...
+import 'package:vector_db/src/collection.dart';
+import 'package:vector_db/vector_db.dart';
 
 class VectorDB {
   VectorDB._internal();
 
   static final VectorDB _shared = VectorDB._internal();
-  final Map<String, Collection> _collections = {};
+
+  // factory VectorDB() {
```
Comment (parthbasole99) [inline lib/src/vector_db.dart]: remove, commented code

File: lib/src/vector_db.dart [hunk 2]
```diff
@@ -1,14 +1,20 @@
...
+  //   return _shared;
+  // }
 
   static VectorDB get shared => _shared;
```
Comment (parthbasole99) [inline lib/src/vector_db.dart]: Stick to one implementation of singleton

**Review by parthbasole99 — COMMENTED** [review]
Few minor changes, other than those LGTM.

File: lib/src/vector_db.dart (hunk 1 above)
Comment (DhavalTaunk08) [reply lib/src/vector_db.dart]: Done

File: lib/src/vector_db.dart (hunk 2 above)
Comment (DhavalTaunk08) [reply lib/src/vector_db.dart]: Changed to singleton

**Comment by parthbasole99** [comment]
LGTM

**Pull request merged**